skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldConfigurationDetector.kt

```diff
@@ -66,17 +72,39 @@ class SkaffoldConfigurationDetector(val project: Project) : ProjectComponent {
             val skaffoldFiles: List<VirtualFile> =
                 SkaffoldFileService.instance.findSkaffoldFiles(project)
             if (skaffoldFiles.isNotEmpty()) {
-                val skaffoldRunConfigList: List<RunConfiguration> =
-                    getRunManager(project)
-                        .allConfigurationsList.filter { it is AbstractSkaffoldRunConfiguration }
-                if (skaffoldRunConfigList.isEmpty()) {
+                if (!hasExistingSkaffoldConfigurations()) {
                     // existing Skaffold config files, but no skaffold configurations, prompt
                     showPromptForSkaffoldConfigurations(project, skaffoldFiles[0])
                 }
             }
         }
+
+        // add VFS listener to ensure we track YAML file changes when no Skaffold configurations
+        // exist yet to show prompt for configurations once Skaffold file is created.
+        addVirtualFileListener(object : VirtualFileListener {
+            override fun contentsChanged(event: VirtualFileEvent) {
+                DumbService.getInstance(project).runWhenSmart {
+                    if (event.file.fileType is YAMLFileType &&
+                        SkaffoldFileService.instance.isSkaffoldFile(
+                            event.file
+                        ) && !hasExistingSkaffoldConfigurations()
+                    ) {
+                        // content changed to be a valid Skaffold file, no configurations, prompt
```

Comment:
super nit: // content changed to be a valid Skaffold file, and no configurations exist, prompt